send-emails-from-a-service/README.md

```diff
@@ -0,0 +1,682 @@
+
+```json
+{
+    "appointmentNo": 2,
+    "doctorName": "thomas collins",
+    "patient": "John Doe",
+    "actualFee": 7000,
+    "discount": 20,
+    "discounted": 5600.0,
+    "paymentID": "8458c75a-c8e0-4d49-8da4-5e56043b1a20",
+    "status": "settled"
+}
+```
+
+### Concepts covered
+
+- REST API
+- HTTP Client
+- Email Client
+
+## Develop the application
+
+### Step 1: Set up the workspace
+
+Install [Ballerina Swan Lake](https://ballerina.io/downloads/) and the [Ballerina Swan Lake VS Code extension](https://marketplace.visualstudio.com/items?itemName=wso2.ballerina) on VS Code.
+
+### Step 2: Develop the service
+
+Follow the instructions given in this section to develop the service.
+
+1. Create a new Ballerina project using the `bal` command and open it in VS Code.
+
+```bash
+$ bal new service-orchestration
+```
+
+2. Introduce the source code in files with the `.bal` extension (e.g., the `main.bal` file). 
+
+Import the 
+- `ballerina/email` module to send emails using the SMTP protocol
+- `ballerina/http` module to develop the REST API and define the clients that can be used to send requests to the backend services
+- `ballerina/log` module to log debug/error information for each client request
+
+```ballerina
+import ballerina/email;
+import ballerina/http;
+import ballerina/log;
+```
+
+3. Define six [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen, the URLs of the backend services and for host, username and password of the SMTP client.
+
+```ballerina
+
+configurable int port = 8290;
+configurable string hospitalServicesBackend = "http://localhost:9090";
+configurable string paymentBackend = "http://localhost:9090/healthcare/payments";
+configurable string gmailHost = "smtp.gmail.com";
+configurable string senderAddress = ?;
+configurable string appPassword = ?;
+```
+
+4. Define two [`http:Client` clients](https://ballerina.io/learn/by-example/#http-client) and one [`email:SmtpClient` client](https://ballerina.io/learn/by-example/#email-client) to send requests to the backend services.
+
+```ballerina
+final http:Client hospitalServicesEP = check initializeHttpClient(hospitalServicesBackend);
+final http:Client paymentEP = check initializeHttpClient(paymentBackend);
+final email:SmtpClient smtpClient = check initializeEmailClient();
+
+function initializeHttpClient(string url) returns http:Client|error => new (url);
+
+function initializeEmailClient() returns email:SmtpClient|error => new (gmailHost, senderAddress, appPassword);
+```
+
+The argument to the `new` expression is the URL for the backend service. 
+
+Alternatively, the clients can be initialized directly with `new` expressions, but a separate function is used to aid with testing.
+
+```ballerina
+final http:Client hospitalServicesEP = check new (hospitalServicesBackend);
+final http:Client paymentEP = check new (paymentBackend);
+```
```

Comment:
Shall we make this a Note like Dulaj had suggested? Changed for service orchestration too.